app/components/frozen-funds/table.jsx

```diff
@@ -1,52 +1,94 @@
 import React from 'react';
 import Media from 'react-media';
-
+import PropTypes from 'prop-types';
+import BN from 'bignumber.js';
+import { FormattedMessage } from 'react-intl';
+import { FREEZE_COEF_BY_TIME, FREEZE_COEF_FACTOR, ECHO_ASSET_PRECISION } from '../../constants/global-constants';
 import Avatar from '../avatar';
 
 class FrozenFundsTable extends React.Component {
 
-	renderRow() {
-		return (
-			<React.Fragment>
-				<tr className="line">
-					<td className="amount"><span>10000000000</span> ECHO</td>
-					<td className="account"><Avatar accountName="test" /> <span>LOLOLOLOLOLOLOLOLOLOLOLOLOLOLOLOLOLO</span> </td>
-					<td className="coefficient">3.0</td>
-					<td className="period">3 months</td>
-					<td className="expiration">23 Jan 2020</td>
-				</tr>
-			</React.Fragment>
-		);
+	formatDate(date) {
+		const parsedDate = new Date(Date.parse(date)).toDateString();
+		const d = parsedDate.substring(parsedDate.indexOf(' '));
+		return d;
+	}
+
+	formatAmount(amount) {
+		return new BN(amount).div(10 ** ECHO_ASSET_PRECISION).toString(10);
+	}
+
+	renderRows(frozenBalances) {
+		let frozenRowTemplate;
+		frozenBalances.sort((f1, f2) => Date.parse(f1.unfreeze_time) - Date.parse(f2.unfreeze_time));
+		if (frozenBalances) {
+			frozenRowTemplate = frozenBalances.map((fBalance) => {
+				const period = FREEZE_COEF_BY_TIME[fBalance.multiplier / FREEZE_COEF_FACTOR];
```

Comment:
use map with actual `fBalance.multiplier` keys not with `fBalance.multiplier / FREEZE_COEF_FACTOR` keys